libknet/transport_common.c

```diff
@@ -0,0 +1,140 @@
+#include "config.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <pthread.h>
+#include <math.h>
+#include <sys/epoll.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <malloc.h>
+#include <arpa/inet.h>
+
+#include "libknet.h"
+#include "host.h"
+#include "link.h"
+#include "logging.h"
+#include "common.h"
+#include "transports.h"
+
+#ifdef DEBUG
+/*
+ * Keeping this light (and therefore not thread-safe) as it's
+ * for debugging only
+ */
+const char *_transport_print_ip(const struct sockaddr_storage *ss)
```

Comment:
we already have those functions from common/netutils.c No need to duplicate them :-) and they are thread safe ;)